engine/src/universe.cpp

```diff
@@ -46,116 +46,45 @@
+
 using std::string;
 using std::vector;
 using std::find;
 
+// externs
 extern void CacheJumpStar( bool );
 extern void SortStarSystems( vector< StarSystem* > &ss, StarSystem *drawn );
-StarSystem* GameUniverse::Init( string systemfile, const Vector &centr, const string planetname )
-{
-    static bool js = true;
-    if (js) {
-        js = false;
-        LoadWeapons( VSFileSystem::weapon_list.c_str() );
-        CacheJumpStar( false );
-    }
-    return this->Universe::Init( systemfile, centr, planetname );
-}
-
 extern void bootstrap_first_loop();
+extern bool RefreshGUI();
+extern float rand01();
+extern int timecount;
+extern StarSystem * GetLoadedStarSystem( const char *system );
+extern bool screenshotkey;
+extern int getmicrosleep();
+extern void MakeStarSystem( string file, Galaxy *galaxy, string origin, int forcerandom );
+extern string RemoveDotSystem( const char *input );
+extern StarSystem * GetLoadedStarSystem( const char *file );
 
-void GameUniverse::Init( int argc, char **argv, const char *galaxy )
-{
-    current_cockpit = 0;
-    //Select drivers
-#if defined (__APPLE__)
-    //get the current working directory so when glut trashes it we can restore.
-    char pwd[MAXPATHLEN];
-    getcwd( pwd, MAXPATHLEN );
-#endif
-    GFXInit( argc, argv );
-#if defined (__APPLE__)
-    //Restore it
-    chdir( pwd );
-#endif
-    StartGFX();
-    InitInput();
-
-    hud_camera = Camera();
-
-    //Hasten splash screen loading, to cover up lengthy universe initialization
-    bootstrap_first_loop();
-
-    this->Universe::Init( galaxy );
-}
-
-GameUniverse::GameUniverse( int argc, char **argv, const char *galaxy )
-{
-    this->Init( argc, argv, galaxy );
-}
-
-GameUniverse::GameUniverse() : Universe()
-{}
-
-GameUniverse::~GameUniverse()
-{
-    DeInitInput();
-    GFXShutdown();
-}
-
-//sets up all the stuff... in this case the ships to be rendered
-
-void GameUniverse::SetupCockpits( vector< string >playerNames )
-{
-    for (unsigned int i = 0; i < playerNames.size(); ++i) {
-        cockpit.push_back( NULL );
-        cockpit.back() = new GameCockpit( "", NULL, playerNames[i] );
-    }
-}
-
-void GameUniverse::activateLightMap( int stage )
-{
-    getActiveStarSystem( 0 )->activateLightMap( stage );
-}
-
-Texture* GameUniverse::getLightMap()
-{
-    return getActiveStarSystem( 0 )->getLightMap();
-}
 
-void GameUniverse::StartGFX()
-{
-    GFXBeginScene();
-    GFXMaterial mat;
-    setMaterialAmbient( mat, 1.0, 1.0, 1.0, 1.0 );
-    setMaterialDiffuse( mat, 1.0, 1.0, 1.0, 1.0 );
-    setMaterialSpecular( mat, 1.0, 1.0, 1.0, 1.0 );
-    setMaterialEmissive( mat, 0.0, 0.0, 0.0, 1.0 );
-    mat.power = 60.0F;
-    unsigned int tmp;
-    GFXSetMaterial( tmp, mat );
-    GFXSelectMaterial( tmp );
-    int ligh;
-    GFXCreateLightContext( ligh );
-    GFXSetLightContext( ligh );
-    GFXLightContextAmbient( GFXColor( 0, 0, 0, 1 ) );
-    GFXEndScene();
-}
+// Local Variables
+vector< StarSystem* >deleteQueue;
```

Comment:
Just FYI - another good refactor would be to take things like `vector< StarSystem*>` and make them types:
```
typedef vector<StarSystem*> StarSystemList;
```
and then use that new type everywhere instead of the more expansive form. This will also help eliminate bugs, but also make typing easier.

Not asking for this round since it's copy/paste moves; but noting as something to do for another round at some point.

Reply:
https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/290 will track this kind of thing.